client.go

```diff
@@ -0,0 +1,436 @@
+package client
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"regexp"
+	"strconv"
+	"strings"
+	"time"
+)
+
+var emailValid = regexp.MustCompile(`^[a-z0-9._%+\-]+@[a-z0-9.\-]+\.[a-z]{2,4}$`)
+
+// InvalidEmail Error
+type InvalidEmail struct{}
+
+func (e InvalidEmail) Error() string {
+	return "Invalid Email format"
+}
+
+// RoleMissing Error
+type RoleMissing struct{}
+
+func (e RoleMissing) Error() string {
+	return "Missing Role Parameter"
+}
+
+// NameMissing Error
+type NameMissing struct{}
+
+func (e NameMissing) Error() string {
+	return "Missing Name Parameter"
+}
+
+// EmailMissing Error
+type EmailMissing struct{}
+
+func (e EmailMissing) Error() string {
+	return "Missing Email Parameter"
+}
+
+// LoginMissing Error
+type LoginMissing struct{}
+
+func (e LoginMissing) Error() string {
+	return "Missing Login Parameter"
+}
+
+// PasswordMissing Error
+type PasswordMissing struct{}
+
+func (e PasswordMissing) Error() string {
+	return "Missing Password Parameter"
+}
+
+// NotInteger Error
+type NotInteger struct{}
+
+func (e NotInteger) Error() string {
+	return "ID provided is not Integer"
+}
+
+// TokenExpired Error
+type TokenExpired struct {
```

Comment:
This error should be handled within the client.
If token expired, new one should be issued.
So if you needed it just remane to tokenExpiredError (lower case)